web3c/confidential_provider.js

```diff
@@ -136,7 +155,12 @@ class ConfidentialSendTransform {
     let tryAdd = outstanding.indexOf(payload.params[0]) > -1;
     return this.provider[this.provider.sendAsync ? 'sendAsync' : 'send'](payload, async (err, res) => {
       if (!err && res.result && res.result.logs && res.result.logs.length) {
-        res.result.logs = await this.tryDecryptLogs(res.result.logs, tryAdd);
+        try {
+          res.result.logs = await this.tryDecryptLogs(res.result.logs, tryAdd)
+        } catch (err) {
+          res = null;
```

Comment:
we should still get the encrypted result even if the logs weren't for us.

Reply:
The problem, alas, is that if we don't set `res = null`, web3.js will just ignore the error.

Reply:
Done. To facilitate this, I changed `tryDecryptLogs` to return an array of both logs and the error, similar to how go does errors, though I'm not sure if this is idiomatic JS (shrug). Let me know if there's a cleaner way.